pkg/block/block.go

```diff
@@ -126,6 +128,34 @@ func cleanUp(logger log.Logger, bkt objstore.Bucket, id ulid.ULID, err error) er
 	return err
 }
 
+// MarkForDeletion creates a file which stores information about when the block was marked for deletion.
+func MarkForDeletion(ctx context.Context, logger log.Logger, bkt objstore.Bucket, id ulid.ULID) error {
+	deletionMarkFile := path.Join(id.String(), metadata.DeletionMarkFilename)
+	deletionMarkExists, err := objstore.Exists(ctx, bkt, deletionMarkFile)
+	if err != nil {
+		return errors.Wrapf(err, "check exists: %s in bucket", deletionMarkFile)
+	}
+	if deletionMarkExists {
+		return errors.Wrapf(err, "file %s already exists in bucket", deletionMarkFile)
```

Comment:
What error are we wrapping here?

This is quite a significant bug as if you do `errors.Wrap(nil, "....."`)` the returned `errors` is also nil... wonder why our tests did not catch it? Can you add a case for that?

In cases like this I like to NOT fix the bug yet. Add case to see the tests failing and ONLY then add fix, WDYT? :hugs:

Reply:
yes, adding a test and then fixing it will help. I added the test.